poradnia/users/views.py

```diff
@@ -11,12 +14,24 @@
 from .utils import PermissionMixin
 
 
-class UserDetailView(PermissionMixin, DetailView):
+class UserDetailView(PermissionRequiredMixin, DetailView):
     model = User
-    # These next two lines tell the view to index lookups by username
     slug_field = "username"
     slug_url_kwarg = "username"
 
+    def handle_no_permission(self):
+        if self.request.user.is_anonymous():
+            return redirect('/konta/login/?next=%s' % self.request.path)
+        else:
+            return HttpResponseForbidden("403 - not authorized")
```

Comment:
To może okazać się interesujące. https://stackoverflow.com/questions/6618939/how-do-i-raise-a-response-forbidden-in-django